api/src/main/java/org/pmiops/workbench/profile/ProfileService.java

```diff
@@ -286,50 +287,132 @@ private void updateInstitutionalAffiliations(Profile updatedProfile, DbUser user
         .orElseThrow(() -> new BadRequestException("Address must not be empty"));
 
-    Address updatedProfileAddress = updatedProfile.getAddress();
+    Address updatedProfileAddress = profile.getAddress();
     if (StringUtils.isEmpty(updatedProfileAddress.getStreetAddress1())
         || StringUtils.isEmpty(updatedProfileAddress.getCity())
         || StringUtils.isEmpty(updatedProfileAddress.getState())
         || StringUtils.isEmpty(updatedProfileAddress.getCountry())
         || StringUtils.isEmpty(updatedProfileAddress.getZipCode())) {
       throw new BadRequestException(
-          "Address cannot have empty street Address 1/city/state/country or Zip Code");
+          "Address cannot have empty street address, city, state, country or zip code");
     }
-    if (updatedProfile.getContactEmail() != null
-        && !updatedProfile.getContactEmail().equals(prevProfile.getContactEmail())) {
-      // See RW-1488.
-      throw new BadRequestException("Changing email is not currently supported");
-    }
-    if (updatedProfile.getUsername() != null
-        && !updatedProfile.getUsername().equals(prevProfile.getUsername())) {
-      // See RW-1488.
-      throw new BadRequestException("Changing username is not supported");
+  }
+
+  private void validateAreaOfResearch(Profile profile) throws BadRequestException {
+    if (StringUtils.isEmpty(profile.getAreaOfResearch())) {
+      throw new BadRequestException("Research background cannot be empty");
     }
   }
 
-  private void validateStringLength(String field, String fieldName, int max, int min) {
-    if (field == null) {
-      throw new BadRequestException(String.format("%s cannot be left blank!", fieldName));
+  /**
+   * Returns a list of PropertyChange entries from a diff that match either a field or any of its
+   * subfields. For example, a pathPrefix of 'address' will match a field change for both 'address'
+   * and for 'address.zipCode'.
+   */
+  private List<Change> getChangesWithPrefix(final Diff diff, final String pathPrefix) {
+    return diff.getChanges(
+        change -> {
+          if (change instanceof PropertyChange) {
+            return ((PropertyChange) change).getPropertyNameWithPath().startsWith(pathPrefix);
+          } else {
+            return false;
+          }
+        });
+  }
+
+  /**
+   * Validates a set of Profile changes by comparing the updated profile to the previous version.
+   * Only fields that have changed are subject to validation.
+   *
+   * <p>If the previous version is null, the updated profile is presumed to be a new profile and all
+   * validation rules are run. If both versions are non-null, only changed fields are validated.
+   *
+   * @param updatedProfile
+   * @param prevProfile
+   * @throws BadRequestException
+   */
+  public void validateProfile(Profile updatedProfile, Profile prevProfile)
+      throws BadRequestException {
+    cleanProfile(updatedProfile);
+
+    boolean isNewObject = prevProfile == null;
```

Comment:
If we could avoid nullable parameters, that would be better. Optional works, but even optional parameters are discouraged.

I think the breakdown is also a bit confusing, because `validateProfile()` calls `cleanProfile()`, which makes changes. I generally expect `validate` methods to just check something and either return a boolean or throw something.